the-new-octal/src/app/service/product.service.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Product } from '../model/product';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+
+  apiUrl: string = `${environment.apiUrl}product`;
+  endPoint: string = 'product';
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  getAll(): Observable<Product[]> {
+    return this.http.get<Product[]>(this.apiUrl);
+  }
+
+  get(id: number): Observable<Product> {
+    return this.http.get<Product>(`${this.apiUrl}/${id}`);
+  }
+
+  update(product: Product): Observable<Product> {
+    return this.http.patch<Product>(`${this.apiUrl}/edit-product/${product.id}`, product);
```

Comment:
A leírásból:
> 3. Szerkesztő komponensek
> Minden entitáshoz (adattípushoz) tartoznak szerkesztőoldalak. Ezek egységesen a következőek szerint működjenek:
> Érdemes a nevüket mindig az edit előtaggal kezdeni a könnyebb azonosítás érdekében, például: edit-product vagy edit-bill.

Tehát a Komponens elérési útvonalába kell szerepeljen az `edit-product`.
Hogyha kipróbálod valami kóddal a futtatást, akkor elméletileg láthatod is, hogy nem jó.